server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -972,6 +972,16 @@ public Object[] addGeneratedValues(IndexItem item) {
         List<Object> extendedValues = new ArrayList<>(insertValues.length);
         Collections.addAll(extendedValues, insertValues);
 
+        // INSERT... ON CONFLICT... UPDATE SET statement might not contain some columns.
+        // Unused columns are anyway added to the targets by the UpdateToInsert since we need all columns to fetch conflicting values.
+        // However, insertValues contains only provided values.
```

Comment:
This looks not correct to me, at https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/execution/dml/upsert/UpdateToInsert.java#L217 each existing columns value will be added to the `IndexItem.insertValues`.

Reply:
I think this link related only to `UPDATE` or update part of the `CONFLICT`, ie part of the logic coming after `tryInsertFirst`.

I highlighted this in https://github.com/crate/crate/pull/15309#issuecomment-1883520075

Reply:
Maybe I need to rephrase comment, 

> However, insertValues contains only provided values.

to 
// However, insertValues contains only provided values on the first phase of the INSERT ON CONFLICT when we try to perform a regular insert

Reply:
> I think this link related only to `UPDATE` or update part of the `CONFLICT`, ie part of the logic coming after `tryInsertFirst`.
> 
> I highlighted this in [#15309 (comment)](https://github.com/crate/crate/pull/15309#issuecomment-1883520075)

Yes this only affects the conflict case of an upsert (no matter how this upsert was originated).
So when you say this is not the problematic logic but the tryInsertFirst which is just a normal insert, the issue must happen on `INSERT` stmt without any `ON CONFLICT` clause, right?

Reply:
> the issue must happen on INSERT stmt without any ON CONFLICT clause, right?

On regular INSERT we don't have targets/values size mismatch because we don't have `UpdateToInsert` which can expands targets.

Reply:
More specifically, this issue is not hit even on `ON CONFLICT DO NOTHING` because then `UpdateToInsert` is not created. I think this issue is only `INSERT... ON CONFLICT... UPDATE SET` specific